### ISSUE TITLE
Deep learning applications for predicting pharmacological properties of drugs and drug repurposing using transcriptomic data

### ISSUE THREAD

**Issue by cgreene** [greenelab/deep-review#38]
https://doi.org/10.1021/acs.molpharmaceut.6b00248

**Comment by cgreene** [comment]
Paper on LINCS. May be important for the "treat" discussion. Worth noting that the set restricted to signalling pathways outperforms the full set. Raises some questions about their NN's ability to do feature selection/construction.

**Comment by kumardeep27** [comment]
This work demonstrates the utility of deep neural networks (DNN) for classification of many drugs into therapeutic categories based solely on transcriptomic response profiles. Paper uses classification of 678 drugs for a time series of 6-24 hrs (into 12 categories) tested across 3 cell lines (A549, MCF-7 and PC-3) available from LINCS resource. Base dataset for classification rely upon the trivial gene expression and processed pathway activation scoring system. DNN outperformed SVM and within DNN, pathway level models are better in performance.

Level 3 (Q2NORM(preprocessed and prenormalized)) expression data of 12,797 genes was used as transcriptomic data for 3 cell lines. MeSH classification was used to categorize drugs (each drug belonging to single class) into 12 categories. Cell line, perturbation concentration and time were used to select 26,420 drugs' samples. Total drug perturbation samples were 26,420 for 3 cell lines. Using full transcriptomic space of 12,797 genes, the mean F1 score for 12 categories was 0.24. This led to the implementation of novel feature spaces in terms of pathway activation scoring and landmark genes.

Pathway analysis is considered as biologically meaningful counterpart of PCA  for feature selection by data projection. DNN like other DL tools have the ability to handle sparse as well as complex data structures. For each case sample group, a reference group (treated with DMSO) was generated. Differentially expressed genes were identified with gold-standard parameters and OncoFinder was used to get the qualitative measure of pathway activation known as pathway activation strength (PAS). Pathway level features rely on OncoFinder pathway analysis tool capable of calculating quantitative pathway activation scoring (PAS) profiles for 271 signaling pathways for each sample. Samples for which total score was 0 were discarded leaving final 9352 samples with 308, 454 and 433 drugs for A549, MCF7 and PC3 cell lines respectively.

For SVM, nested CV of 3-fold was used with different kernel and cost function inside the outer 10-fold CV settings.
DNN was with standard MLP settings and 977 gene expression nodes and 271 PAS nodes. Grid search was used for parameter optimization, similar to the SVM protocol. cross entropy as cost function and AdaDelta as cost function optimizer was used. After experimenting with a number of parameters setting, authors used the 3 hidden layers, 200 nodes in each and rectified linear activation function.
The code is available on github.

Multiclass approach was done for 3-,5- and 12 classes with 10-fold CV using DNN framework. Performance for all the approaches were better for DNN as compared to SVM. Antineoplastic class was top performer. 
977 Landmark genes (covering most of the variation) were selected based on LINCS authors' observations. DNN was applied to same number of samples to compare the results with pathway based approach. Figure 2 shows that DNN is better than SVM and Pathway based features outperformed Landmark gene based features. 
Since the study is multiclass problem, so there were also cases of misclassification. Antineoplastic agents were the most correctly predicted drugs with 0.686 accuracy on 12 classes. Figure 3 shows the confusion matrix of DNN classification and misclassification. Latter was suggested as a boon for the drug repurposing e.g. cardio drugs as CNS drugs.
They show results on 'landmark gene' level and 'pathway' level and showed the latter one to be more efficient.

**Comment by agitter** [comment]
@kumardeep27 Thanks for another summary.  Can you say a bit more about the overfitting concerns?

**Comment by kumardeep27** [comment]
In Figure 2 (d,e,f), for the DNN algorithm the average F1 score for 10-fold has much difference between the training and validation dataset as compared to the figure (a,b,c).
The performance on the validation set should be comparable to that of the training performance.
